### PR TITLE
Add timestamp to published message

### PR DESCRIPTION
**Pull request by edgarlatorre**
### :tophat: What?

* Add optional parameter published_at  to publish function, it uses `time.time()` in case published_at is None.
* client.Publisher documentation page.

### Screenshot

Screenshot 2019-06-11 at 14 46 56

### :thinking: Why?

It can be useful to have the time that the message was published.

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -1,43 +1,64 @@
...
+from unittest.mock import ANY, MagicMock, patch
 
 from google.cloud.pubsub_v1 import PublisherClient
 
 from rele import Publisher
 
 
 class TestPublisher:
+    @classmethod
+    def setup_method(cls):
```
Comment (tobami) [inline tests/test_client.py]: We usually create a class-scoped fixture for this instead

Comment (edgarlatorre) [reply tests/test_client.py]: Done.

**Comment by andrewgy8** [comment]
Addresses #31

File: rele/client.py [hunk 2]
```diff
@@ -49,6 +50,22 @@ def __init__(self):
                 credentials=settings.RELE_GC_CREDENTIALS)
 
     def publish(self, topic, data, blocking=False, **attrs):
+        """Publish data to Pub/Sub.
+
```
Comment (andrewgy8) [inline rele/client.py]: I would suggest to add a sentence about the `published_at` time stamp attribute that we add.

Comment (edgarlatorre) [reply rele/client.py]: @andrewgy8 I've changed the doc with your suggestion. Let me know if it's ok.

File: rele/client.py [hunk 3]
```diff
@@ -49,6 +50,22 @@ def __init__(self):
...
+            publisher.publish('topic_name', {'foo': 'bar'})
+
+        :param topic: string topic to publish the data.
+        :param data: dict with the content of the message.
+        :param blocking: boolean, default False.
+        :param attrs: Extra parameters to be published.
+        :return: future
+        """
+
+        attrs['published_at'] = str(time.time())
```
Comment (tobami) [inline rele/client.py]: Since casting to string is needed, shouldn't we do the opposite on the consuming side?

Comment (edgarlatorre) [reply rele/client.py]: If every attribute in the message has to be a string I don't see the point of doing this. Also it should be implemented by the developer that is writing the subscription, isn't it?

Comment (andrewgy8) [reply rele/client.py]: I think what @tobami is suggesting is that we deserialize the message data from bytes to a python dictionary. So we may want to serialize from a string timestamp to a python datetime object. Is that right?

Comment (tobami) [reply rele/client.py]: Yep, same as we don't make the user decode and deserializer every payload, published_at has a very specific use case, so we'll always want it in the right type

Comment (edgarlatorre) [reply rele/client.py]: Okay, just implemented it.

Comment (edgarlatorre) [reply rele/client.py]: @tobami @andrewgy8 Can I merge it?

File: tests/test_client.py [hunk 4]
```diff
@@ -1,46 +1,51 @@
...
-
-from . import settings
+from unittest.mock import ANY, patch
 
 
+@pytest.mark.usefixtures('publisher')
 class TestPublisher:
-
-    def test_returns_future_when_published_called(self):
+    @patch('time.time', return_value=1560244246.863829)
```
Comment (jonasae) [inline tests/test_client.py]: We use to do this kind of patching (datetime, time) using a pytest fixture.

For instance:
```
@pytest.fixture
def published_at():
  return 1560244246.863829
@pytest.fixture
def time_mock(published_at):
  with patch('time.time') as mock:
    mock.return_value = published_at
    yield mock
```
Later on, as mock is not going to be referred to, but `published_at` will do:
```
@pytest.mark.usefixtures('publisher, time_mock')
def test(self, published_at, publisher):
...
assert_called_with(... published_at=published_at ...)
```
Furthermore, both fixtures will be very useful and reusable.

Comment (edgarlatorre) [reply tests/test_client.py]: Cool, I'm gonna do that. Thanks!

**Pull request merged**